### PR TITLE
Db models

### PR DESCRIPTION
**Pull request by Grotawred**
Снимок экрана 2023-06-17 233924
Spring Serucity not working but it is without errors

### CODE REVIEW

File: src/main/java/com/example/demo/service/UserService.java [hunk 1]
```diff
@@ -38,18 +41,23 @@ public User registerUser(RegistrationRequest request) {
...
     return userRepository.save(mapper.userDtoToUser(newUser));
   }
 
   @Override
   public List<UserDTO> findByEmail(String email) {
-    return mapper.listOfUserToListOfUserDto(userRepository.findByEmail(email));
+    List<UserDataDTO> userData = mapper.listOfUserDataToListOfUserDataDto(userDataRepository.findByEmail(email));
+    List<User> user;
+    if (userData.isEmpty()){
+      return null;
```
Comment (vovik541) [inline src/main/java/com/example/demo/service/UserService.java:56]: try to avoid returning nulls. It is better to return Optional or at leas empty object

**Review by vovik541 — CHANGES_REQUESTED** [review]

**Comment by Grotawred** [comment]
Снимок экрана 2023-06-19 202731

File: src/main/java/com/example/demo/model/Country.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class Country {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "county_id")
+    private Long id;
+    @OneToOne
+    @JoinColumn(name = "telephone_code_id")
+    private TelephoneCode telephoneCode;
+    @Column(name = "name")
+    private String country;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/Country.java:23]: name

File: src/main/java/com/example/demo/model/PersonalData.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private Long id;
+    @OneToOne
+    @JoinColumn(name = "telephone_id")
+    private Telephone telephone;
+    @OneToOne
+    @JoinColumn(name = "post_id")
+    private Post post;
+    @ManyToMany
+    @JoinTable(
+            name = "personal_page",
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/PersonalData.java:30]: personal_data_paragraphs_relations

File: src/main/java/com/example/demo/model/Telephone.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "telephone_id")
+  private Long id;
+
+  @OneToOne
+  @JoinColumn(name = "telephone_code_id")
+  private TelephoneCode telephoneCode;
+
+  @Column(name = "telephone")
+  private int telephoneNumber;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/Telephone.java:22]: check type

File: src/main/java/com/example/demo/model/User.java [hunk 5]
```diff
@@ -1,31 +1,39 @@
...
+          joinColumns = @JoinColumn(name = "user_id"),
+          inverseJoinColumns = @JoinColumn(name = "role_id"))
+  private Set<Role> roles;
+  @OneToOne
+  @JoinColumn(name = "personal_data_id", referencedColumnName = "personal_data_id")
+  private PersonalData personalData;
+  @Column(name = "is_verified")
+  private boolean isVerified;
+  @Column(name = "registration_date")
+  private LocalDateTime registrationDate;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/User.java:31]: registrationDate -> registrationDateTime

File: src/main/java/com/example/demo/model/User.java [hunk 6]
```diff
@@ -1,31 +1,39 @@
...
+  @Column(name = "verification_date")
+  private LocalDateTime verificationDate;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/User.java:33]: the same

File: src/main/java/com/example/demo/model/dto/CountryDTO.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import com.example.demo.model.TelephoneCode;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class CountryDTO {
+    private Long id;
+    private TelephoneCode telephoneCode;
+    private String country;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/dto/CountryDTO.java:12]: name

File: src/main/java/com/example/demo/model/dto/PersonalDataDTO.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+@Data
+@Builder
+public class PersonalDataDTO {
+    private Long id;
+    private Telephone telephone;
+    private Post post;
+    private Set<Paragraph> paragraphs;
+    private String email;
+    private String firstName;
+    private String last_name;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/dto/PersonalDataDTO.java:20]: last_name -> lastName

File: src/main/java/com/example/demo/model/dto/TelephoneCodeDTO.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package com.example.demo.model.dto;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class TelephoneCodeDTO {
+    private Long id;
+    private int code;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/dto/TelephoneCodeDTO.java:10]: String

File: src/main/java/com/example/demo/model/dto/TelephoneDTO.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import com.example.demo.model.TelephoneCode;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class TelephoneDTO {
+    private Long id;
+    private TelephoneCode telephoneCode;
+    private int telephoneNumber;
```
Comment (vovik541) [inline src/main/java/com/example/demo/model/dto/TelephoneDTO.java:12]: int ?

File: src/main/java/com/example/demo/security/UserRegistrationDetails.java [hunk 11]
```diff
@@ -13,17 +13,17 @@
 @Data
 public class UserRegistrationDetails implements UserDetails {
 
-  private String userName;
+  private String username;
```
Comment (vovik541) [inline src/main/java/com/example/demo/security/UserRegistrationDetails.java:16]: userName -> login

File: src/main/java/com/example/demo/security/UserRegistrationDetailsService.java [hunk 12]
```diff
@@ -17,9 +17,9 @@ public class UserRegistrationDetailsService implements UserDetailsService {
   private final UserMapper mapper;
 
   @Override
-  public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
     return mapper.listOfUserToListOfUserDto(userRepository
-                    .findByEmail(email)).stream().findAny()
+                    .findByLogin(username)).stream().findAny()
```
Comment (vovik541) [inline src/main/java/com/example/demo/security/UserRegistrationDetailsService.java:22]: !!!

File: src/main/java/com/example/demo/service/UserService.java [hunk 13]
```diff
@@ -32,24 +39,33 @@ public List<UserDTO> getUsers() {
   @Override
   public User registerUser(RegistrationRequest request) {
     List<UserDTO> user = this.findByEmail(request.email());
-    if (user.isEmpty()) {
+    if (!user.isEmpty()) {
       throw new UserAlreadyExistsException(
           TEXT_USER_WITH_EMAIL + request.email() + TEXT_ALREADY_EXIST);
```
Comment (vovik541) [inline src/main/java/com/example/demo/service/UserService.java:44]: !!!

File: src/main/java/com/example/demo/service/UserService.java [hunk 14]
```diff
@@ -32,24 +39,33 @@ public List<UserDTO> getUsers() {
...
+            .roles(roleRepository.findRoleByName(request.role()))
+            .personalData(newUserData)
+            .registrationDate(java.time.LocalDateTime.now())
             .build();
-    return userRepository.save(mapper.userDtoToUser(newUser));
+    return userRepository.save(newUser);
   }
 
   @Override
   public List<UserDTO> findByEmail(String email) {
```
Comment (vovik541) [inline src/main/java/com/example/demo/service/UserService.java:60]: Optional

File: src/main/java/com/example/demo/service/UserService.java [hunk 15]
```diff
@@ -32,24 +39,33 @@ public List<UserDTO> getUsers() {
...
-    return mapper.listOfUserToListOfUserDto(userRepository.findByEmail(email));
+    PersonalDataDTO userData = mapper.personalDataToPersonalDataDto(userDataRepository.findByEmail(email));
+    List<User> user;
+    if (userData == null) {
```
Comment (vovik541) [inline src/main/java/com/example/demo/service/UserService.java:63]: isNull / notNull

File: src/main/java/com/example/demo/service/UserService.java [hunk 16]
```diff
@@ -62,15 +78,15 @@ public void saveUserVerificationToken(UserDTO theUser, String token) {
   public String validateToken(String theToken) {
     VerificationToken token = tokenRepository.findByToken(theToken);
     if (token == null) {
-      return TEXT_ABOUT_INVALID_VERIFICATION_TOKEN;
+      return INVALID_VERIFICATION_TOKEN_LOG;
```
Comment (vovik541) [inline src/main/java/com/example/demo/service/UserService.java:81]: _MESSAGE

**Review by vovik541 — CHANGES_REQUESTED** [review]

**Pull request merged**